x-pack/plugins/enterprise_search/server/routes/enterprise_search/search.ts

```diff
@@ -64,21 +67,33 @@ export function registerSearchRoute({ router, log }: RouteDependencies) {
       const { client } = (await context.core).elasticsearch;
       const { page = 0, size = ENTERPRISE_SEARCH_DOCUMENTS_DEFAULT_DOC_COUNT } = request.query;
       const from = page * size;
-      const searchResults: SearchResponseBody = await fetchSearchResults(
-        client,
-        indexName,
-        searchQuery,
-        from,
-        size
-      );
+      try {
+        const searchResults: SearchResponseBody = await fetchSearchResults(
+          client,
+          indexName,
+          searchQuery,
+          from,
+          size
+        );
 
-      return response.ok({
-        body: {
-          meta: calculateMeta(searchResults, page, size),
-          results: searchResults,
-        },
-        headers: { 'content-type': 'application/json' },
-      });
+        return response.ok({
+          body: {
+            meta: calculateMeta(searchResults, page, size),
+            results: searchResults,
+          },
+          headers: { 'content-type': 'application/json' },
+        });
+      } catch (error) {
+        if (isIndexNotFoundException(error)) {
+          return createError({
+            errorCode: ErrorCode.INDEX_NOT_FOUND,
+            message: 'Could not found index',
```

Comment:
whilst its not everywhere, we typically localise the error message here, especially if presented to the user. Also message could be better: "Index does not exist".